### PR TITLE
Create the folders for the Sqlite blob files if they do not exist.

### PR DESCRIPTION
**Pull request by seriouz**
As mentioned in bagetter/BaGetter#38 under some special configurations Sqlite failed with a "folder not found" exception.
This PR solves this issue. It checks the config and creates the folder under Windows or *nix like OS during the database migration respecting full qualified path or relative paths.

Addresses https://github.com/bagetter/BaGet/issues/38

### CODE REVIEW

**Review by daghsentinel — APPROVED** [review]

**Comment by seriouz** [comment]
@daghsentinel Thanks for the review! I'll wait for another review before merging this.

File: src/BaGetter.Database.Sqlite/SqliteContext.cs
Comment (Regenhardt) [inline src/BaGetter.Database.Sqlite/SqliteContext.cs:120]: Is the inner `Path.Combine` necessary? What does it even do on the output of a previous `Path.Combine`?

**Review by Regenhardt — CHANGES_REQUESTED** [review]
Ok there has to be a better way to do this, like telling EFCore to create the directory if it doesn't exist, or at least have .NET handle the OS-specifics. We're on .NET 8, we shouldn't have to handle different directory structures ourselves any more.

**Comment by seriouz** [comment]
> Ok there has to be a better way to do this, like telling EFCore to create the directory if it doesn't exist, or at least have .NET handle the OS-specifics. We're on .NET 8, we shouldn't have to handle different directory structures ourselves any more.

This point baffled me too. It seems that the implementation of EFCore Sqlite expects the folder for the database to be created before hand. EFCore will create the files, yes but not the folders (afaik).

File: src/BaGetter.Database.Sqlite/SqliteContext.cs
Comment (seriouz) [reply src/BaGetter.Database.Sqlite/SqliteContext.cs:120]: I'll check if this makes a difference

**Comment by FroggieFrog** [comment]
Maybe you can be more fancy and use source generators for the Regex:

```cs
    public partial class SqliteContext : AbstractContext<SqliteContext>
    {
        [GeneratedRegex("^[A-Za-z]:")]
        private static partial Regex WindowsDriveLetterRegex();
```

Also I think it could be beneficial to add [TestableIO.System.IO.Abstractions](https://github.com/TestableIO/System.IO.Abstractions). This would make it easier to write some tests for this.

**Comment by seriouz** [comment]
I've tested around with root and relativ pathes under Windows and Linux and @Regenhardt you were kind of right; I simplified the logic dramatically.

File: src/BaGetter.Database.Sqlite/SqliteContext.cs
Comment (Regenhardt) [inline src/BaGetter.Database.Sqlite/SqliteContext.cs:88]: I'm not sure `exeDir` is the best description here as this will be the directory it's executed from, not the directory the executable is in.

File: src/BaGetter.Database.Sqlite/SqliteContext.cs
Comment (Regenhardt) [inline src/BaGetter.Database.Sqlite/SqliteContext.cs:77]: Why is this split even necessary? As far as I can see in a quick test, `Directory.CreateDirectory` will create both absolute and relative paths you throw at it, starting at the present working directory.

**Review by Regenhardt — CHANGES_REQUESTED** [review]

**Comment by Regenhardt** [comment]
Definitely much better already :)

File: src/BaGetter.Database.Sqlite/SqliteContext.cs
Comment (seriouz) [reply src/BaGetter.Database.Sqlite/SqliteContext.cs]: Thanks for looking into this. I've tested again with your suggested changes and it worked flawlessly. I don't know what setup i had (under Windows) where it did not work so i added the switching.
The CreateDirectory() implementation keeps track of relative or root paths itself and worked without problems under Linux and Windows.

File: src/BaGetter.Database.Sqlite/SqliteContext.cs
Comment (seriouz) [reply src/BaGetter.Database.Sqlite/SqliteContext.cs]: Removed the unecessary function CreateFullDataSourcePath.

**Review by Regenhardt — APPROVED** [review]

**Pull request merged**
